addons/survey/static/src/js/survey_form.js

```diff
@@ -1196,6 +1199,9 @@ publicWidget.registry.SurveyFormWidget = publicWidget.Widget.extend(SurveyPreloa
     _resetErrors: function () {
         this.$('.o_survey_question_error').empty().removeClass('slide_in');
         this.$('.o_survey_error').addClass('d-none');
+        for (const row of this.el.querySelectorAll('.o_survey_question_matrix > tbody > tr > th.bg-danger')) {
```

Comment:
Maybe simplify the selector a bit ?

`.o_survey_question_matrix th.bg-danger` ?

Should be safe enough.

Also could you use a forEach as well for consistency? Seems like you used forEach above.